File: Office-Hours/Relative-Path-Python/tests/test_path_demo.py
import os

from hamcrest import assert_that, equal_to


def test_relative_path() -> None:
    path = os.path.join("resources", "testProgram1", "KeyboardInput.java")

    with open(path, "r") as a_file:
        first_line = a_file.readline()
        assert_that(first_line, equal_to("import java.util.Scanner;"))

File: Office-Hours/Relative-Path-Rust/tests/test_path_demo.rs
use std::fs::File;
use std::io::BufReader;
use std::io::prelude::*;
use std::path::PathBuf;

use hamcrest2::prelude::*;
use rstest::*;

#[rstest]
pub fn test_relative_path() {
    let path: PathBuf = ["resources", "testProgram1", "KeyboardInput.java"]
        .iter()
        .collect();

    let first_line = {
        let file = File::open(path).unwrap();
        let mut reader = BufReader::new(file);

        let mut line = String::new();
        reader.read_line(&mut line).unwrap();

        line
    };

    assert_that!(first_line, equal_to("import java.util.Scanner;"));
}
